src/utils.ts

```diff
@@ -31,6 +31,24 @@ export function extractFilename(path: string): string {
   }
 }
 
+export function decodeStage(x: string, y: string) {
+  // If file is untracked
+  if (x === '?' && y === '?') {
+    return 'untracked';
+  } else {
+    // If file is staged
+    if (x !== ' ' && y !== 'D') {
+      return 'staged';
+    }
+    // If file is unstaged but tracked
+    if (y !== ' ') {
+      return 'unstaged';
+    }
+  }
+
+  return undefined;
```

Comment:
I second this advice. We should avoid explicitly setting variables to `undefined` (or, more generally, to its equivalent `void 0`).

Reply:
Done. It made sense at the time since one of my use cases was within the composition of a Component signature:

```
https://github.com/jupyterlab/jupyterlab-git/pull/448/files#diff-e9db4aac31cba6a3cbd3b59121720955R81-R88
```

but I see your point